libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go

```diff
@@ -66,45 +61,177 @@ func (fetcher GethRpcStorageFetcher) FetchStorageDiffs(out chan<- types.RawDiff,
 		case err := <-clientSubscription.Err():
 			logrus.Errorf("error with client subscription: %s", err.Error())
 			errs <- err
-		case diff := <-ethStatediffPayloadChan:
-			logrus.Trace("received a statediff")
-			stateDiff := new(statediff.StateDiff)
-			decodeErr := rlp.DecodeBytes(diff.StateDiffRlp, stateDiff)
-			logrus.Tracef("received a statediff from block: %v", stateDiff.BlockNumber)
-			if decodeErr != nil {
-				logrus.Warn("Error decoding state diff into RLP: ", decodeErr)
-				errs <- decodeErr
+		case diffPayload := <-ethStatediffPayloadChan:
+			logrus.Trace("received a statediff payload")
+
+			switch fetcher.gethVersion {
+			case OldGethPatch:
```

Comment:
Distinguishing between the different geth patches is confusing, so I'm definitely up for other ideas on how we want to do this. Also my thought it that supporting 3 versions of geth is short-term, and we'll likely deprecate `OldGethPatch` and `NewGethPatchWithService` at some point in favor of `NewGethPatchWithFilter. But again, I'm totally open for other ideas on this!